go/pkg/filemetadata/filemetadata.go

```diff
@@ -0,0 +1,41 @@
+// Package filemetadata contains types of metadata for files, to be used for caching.
+package filemetadata
+
+import (
+	"os"
+	"time"
+
+	"github.com/bazelbuild/remote-apis-sdks/go/digest"
+)
+
+// Metadata contains details for a particular file.
+type Metadata struct {
+	Digest       digest.Digest
+	IsExecutable bool
+	MTime        time.Time
+}
+
+// Compute computes a Metadata from a given file path.
+func Compute(filename string) (*Metadata, error) {
+	file, err := os.Stat(filename)
+	if err != nil {
+		return nil, err
+	}
+	dg, err := digest.NewFromFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	return &Metadata{
+		Digest:       dg,
+		IsExecutable: (file.Mode() & 0100) != 0,
+		MTime:        file.ModTime(),
+	}, nil
+}
+
+// NoopFileMetadataCache is a non-caching metadata cache (always computes).
+type NoopFileMetadataCache struct{}
+
+// Get computes the metadata from the file contents.
+func (c *NoopFileMetadataCache) Get(path string) (*Metadata, error) {
```

Comment:
The cache currently returns a digest, not a pointer to the digest. So make this return Metadata?

Reply:
That's actually intentional -- remember we said that we always pass all structs by pointer, with the sole exception of Digest? I want to follow that, just for simplicity.